cpp/ScanChain_Zp.C

```diff
@@ -649,6 +675,36 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process) {
       double puppimet_pt  = puppimet.first;
       double puppimet_phi = puppimet.second;
 
+      // Account for tunepRelPt if useTuneP
+      vector<LorentzVector> Muon_p4 = {};
+      vector<double> Muon_pt = {};
+      vector<double> Muon_tkRelIso = {};
+      TVector2 pfmet_temp;
+      pfmet_temp.SetMagPhi(pfmet_pt,pfmet_phi);
+      for ( unsigned int im=0; im < nt.nMuon(); im++ ) {
+	Muon_p4.push_back(LorentzVector(nt.Muon_pt().at(im),nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im)));
+	Muon_pt.push_back(nt.Muon_pt().at(im));
+	Muon_tkRelIso.push_back(nt.Muon_tkRelIso().at(im));
+	if ( nt.Muon_highPtId().at(im)>=2 ) {
+	  if ( useTuneP ) {
+	    TVector2 muon_temp;
+	    TVector2 muon_temp_tunep;
+	    muon_temp.SetMagPhi(Muon_pt[im],nt.Muon_phi().at(im));
+	    Muon_p4[im] = LorentzVector(Muon_pt[im],nt.Muon_eta().at(im),nt.Muon_phi().at(im),nt.Muon_mass().at(im));
+	    Muon_pt[im] *= nt.Muon_tunepRelPt().at(im);
+	    muon_temp_tunep.SetMagPhi(Muon_pt[im],nt.Muon_phi().at(im));
+	    // Correct PF MET for highPtId muon candidates
+	    pfmet_temp+=muon_temp;
+	    pfmet_temp-=muon_temp_tunep;
+	  }
+	  else {
+	    Muon_tkRelIso[im] *= nt.Muon_tunepRelPt().at(im);
+	  }
+	}
+      }
+      pfmet_pt  = pfmet_temp.Mod();
```

Comment:
If we are using PUPPI MET, shouldn't we do the same for the PUPPI MET variables as well? Because, in essence, PUPPI MET is constructed by the PF candidates, after having their pT reweighted by the PUPPI weight, right?

Also, if we do so, I guess we are assuming that PUPPI weight = 1 for the highPtId muons that we select as candidates (probably a decent assumption).

Reply:
Good point. As there was no recommendation against using PUPPI MET with highPtId muons I haven't done it. But can be done also for PUPPI MET.

Reply:
Done.